### ISSUE TITLE
Incorrect mouse positioning on high DPI display

### ISSUE THREAD

**Issue by whispy** [rdp/screen-capture-recorder-to-video-windows-free#56]
The mouse is not positioned correctly when recording on high DPI displays.

Let me know if you need more info :)

**Comment by rdp** [comment]
what os?

**Comment by whispy** [comment]
Windows 8 on a Surface Pro 3. Using screen-capture-recorder via ShareX's
interface.

**Comment by mekwall** [comment]
Is this being looked into? It's an issue for everyone on Windows using any other DPI setting than default.

**Comment by whispy** [comment]
@rdp @mekwall I should mention that I figured out a kludgy way to get the mouse positioned correctly when recording in ShareX, but it doesn't use screen-capture-recorder. If I select 'GDI Grab' as the video source and then go to ShareX\Tools\, I can access ffmpeg.exe's settings and check 'Disable display scaling on high DPI settings'.

**Comment by rdp** [comment]
I could look at it with some funding I suppose :|

**Comment by jgbecker** [comment]
@whispy Thank you very much for your providing a solution. It works perfectly.

**Comment by kujovic** [comment]
> If I select 'GDI Grab' as the video source and then go to ShareX\Tools, I can access ffmpeg.exe's settings and check 'Disable display scaling on high DPI settings'.

@whispy @jgbecker Where did you find these ffmpeg settings exactly? I couldn't find them and I looked everywhere, maybe I'm missing something. My ShareX\Tools folder only contains ffmpeg.exe. I tried configuring it via command line but couldn't find any DPI setting.

Also I couldn't find any ffmpeg/DPI settings in the ShareX interface itself and changing video source to GDI Grab does nothing. Could you please explain in more detail how to get to those settings?

**Comment by whispy** [comment]
@kujovic, I wasn't clear — the settings are found in the `Properties` of the `ffmpeg.exe` file itself. If you open Windows Explorer and navigation to your `ShareX\Tools` folder (wherever it is located on your computer), you'll see `ffmpeg.exe` in that folder. Right-click it, select `Properties`, then go to the `Compatibility` tab. A little over halfway down this tab is the `Disable display scaling on high DPI settings` option.

**Comment by kujovic** [comment]
@whispy Very concise, I got it now, thanks! By the way, on Windows 10, after the Anniversary Update, the setting looks like this:

![2017-06-07_22-37-20](https://user-images.githubusercontent.com/2381553/26900269-ecee488a-4bd1-11e7-8d67-c5e426beefae.png)

The first option (`Application`) needs to be selected, the other two don't fix the issue. But... This is how the cursor looks like after applying that option:

![2017-06-07_22-27-06](https://user-images.githubusercontent.com/2381553/26900647-3123ca74-4bd3-11e7-8424-533e6ce6e911.png)
It's enlarged and horrifying! You can compare it with the cursor in the previous screenshot. I don't even know where could this kind of distortion come from. Does your cursor look normal?

**Comment by whispy** [comment]
Yeah, my cursor looks normal. I'm not sure what's causing yours to look
like that. Do you have GDI Grab selected in ShareX, or
screen-capture-recorder?

**Comment by kujovic** [comment]
Yeah, I know you use GDI Grab, probably that's why your cursor is normal. I'm on screen-capture-recorder because GDI Grab crashes for me - I'll look into that further. Thanks.

**Comment by ghiscoding** [comment]
I also have this problem and I don't have `ffmpeg` installed, so my tools folder is empty. The setting I have is set to `GDI grab`, so I can't seem to find where to change the properties. However in the mean time, changing my resolution to 100% fixes it, kinda small on High DPI and so I only use it to grab the animated GIF and change back my resolution to 150%. I'm on Windows 10, if anyone else find a better work around please let us know

**Comment by AzzyC** [comment]
https://user-images.githubusercontent.com/36119634/107558416-a5914f80-6bd2-11eb-9414-f02cf46cc190.mp4

From Command Line:
reg add "HKCU\Software\Microsoft\Windows NT\CurrentVersion\AppCompatFlags\Layers" -v "C:\Users\USERNAME\Documents\ffmpeg\ffmpeg.exe" -t REG_SZ -d HIGHDPIAWARE -f

**Comment by llealloo** [comment]
AzzyC's solution fixed it for me, easy fix thank you!

**Comment by LoganDark** [comment]
> @rdp @mekwall I should mention that I figured out a kludgy way to get the mouse positioned correctly when recording in ShareX, but it doesn't use screen-capture-recorder. If I select 'GDI Grab' as the video source and then go to ShareX\Tools, I can access ffmpeg.exe's settings and check 'Disable display scaling on high DPI settings'.
> 
> This may also work if screen-capture-recorder is the selected video source, but I have not tested it.

@whispy This worked perfectly, it fixed all my problems, including cursor positioning _and_ cursor size :heart:

**Comment by doneykoo** [comment]
> reg add "HKCU\Software\Microsoft\Windows NT\CurrentVersion\AppCompatFlags\Layers" -v "C:\Users\USERNAME\Documents\ffmpeg\ffmpeg.exe" -t REG_SZ -d HIGHDPIAWARE -f

and

> 'GDI Grab' as the video source

Thanks a lot. It works perfectly.

And,
* We can change the `C:\Users\USERNAME` part to be `%UserProfile%`, so people can directly use the command.
* For some people, in addition, the path might be `"%UserProfile%\Documents\ShareX\Tools\ffmpeg.exe"`, you'd better check your own ShareX - screen recording options - FFmpeg path.

**Comment by pvojtech** [comment]
Hi everyone. The high DPI settings fixed cursor alignment (thankyou @AzzyC and others), but the cursor size remains tiny in the recording (windows 10). Does anyone know how to fix the cursor size?

**Comment by hybridherbst** [comment]
Can confirm, the cursor position is correct but even on 100% scale the cursor size is incorrect in the ffmpeg capture.

**Comment by rdp** [comment]
For some reason I can't even get the mouse to show up at all not sure what I'm missing...
